src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -246,45 +249,36 @@ const ContextualMenu = <L,>({
   useEffect(() => {
     if (visible !== previousVisible) {
       if (visible && !isOpen) {
-        openPortal();
+        handleOpen();
       } else if (!visible && isOpen) {
-        closePortal();
+        handleClose();
       }
     }
-  }, [closePortal, openPortal, visible, isOpen, previousVisible]);
+  }, [handleClose, handleOpen, visible, isOpen, previousVisible]);
 
-  const onResize = useCallback(
-    (evt) => {
-      const parent = getPositionNode(wrapper.current, positionNode);
-      if (parent && !getPositionNodeVisible(parent)) {
-        // Hide the menu if the item has become hidden. This might happen in
-        // a responsive table when columns become hidden as the page
-        // becomes smaller.
-        closePortal(evt);
-      } else {
-        // Update the coordinates so that the menu stays relative to the
-        // toggle button.
-        updatePositionCoords();
-      }
-    },
-    [closePortal, positionNode, updatePositionCoords]
-  );
+  useOnClickOutside(wrapper, handleClose, {
+    isEnabled: closeOnOutsideClick,
+  });
```

Comment:
The `isEnabled` flag seems a bit unnatural to have in the hook.
We can let the client do that: `useOnClickOutside(wrapper, closeOnOutsideClick ? handleClose : () => {});`.

Same for the 'useOnEscapePressed` hook, that seemed cleaner without the flag.

Reply:
This is a pattern borrowed from react-query and I believe it's quite common: https://tanstack.com/query/v4/docs/react/guides/disabling-queries